### DISCUSSION TITLE
can we add feature that replace  the selected text into bionic text insted of opening in new page.

### DISCUSSION THREAD

**Discussion by isahillohiya** [kwame-mintah/bionic-speed-reader-browser-extension#8] [Ideas]
can we add feature that replace  the selected text into bionic text insted of opening in new page.

**Comment by kwame-mintah** [comment]
Hi @isahillohiya - this is the next feature coming, which is to allow users to converter the webpage entire page to Bionic Reading.

When this feature is implemented, either the user will be prompted to convert the page to Bionic Reading, or will select the extension and click 'Convert page to Bionic Reading'.

Currently selecting text and choosing the option 'Bionic Reading convert text ...' was a way to allow users to convert specific text to Bionic Reading instead of the whole page.

**Comment by kwame-mintah** [comment]
@isahillohiya - I've made the first attempt of replacing the current page into Bionic Reading, would you be able to pull the latest from master branch and test this feature. It's not perfect at the moment and more changes need to be made.

Please take note how many paragraphs is being converted on the page, as there is a hard limit of 500 requests on the [basic tier ](https://rapidapi.com/bionic-reading-bionic-reading-default/api/bionic-reading1/pricing)of Bionic Readings API:

Screenshot of the current implementation and converting a page to Bionic Reading:
<img width="1289" alt="Screenshot 2022-05-24 at 10 42 12" src="https://user-images.githubusercontent.com/37197235/170002356-5c3f67f4-07d6-4564-8e51-6e1c33e6eb33.png">
